js/interface.js

```diff
@@ -1845,7 +1845,7 @@ function setSelectedTokenDetails(id, name) {
 function getFilteredSpecificTokenList() {
   var rules = _.filter(currentDataSourceRules, function(currentRules) {
     return _.some(currentRules.allow.tokens, function(allowTokenId) {
-      if (widgetData.tokenId) {
+      if (widgetData.tokenId && Number(allowTokenId) !== selectedTokenId) {
```

Comment:
[You mentioned](https://github.com/Fliplet/fliplet-widget-data-source/pull/199/files#r994896009) that `Number()` is no longer needed.

```suggestion
      if (widgetData.tokenId && allowTokenId !== selectedTokenId) {
```

Reply:
Thank you Tony. Done changes